std/concurrency.d

```diff
@@ -2473,6 +2474,23 @@ private
 
 version( unittest )
 {
+    void changeScheduler(Scheduler s)
+    {
+        import core.memory : GC;
+        import core.thread : Thread;
+
+        bool dosleep;
+        while (Thread.getAll().length != 1)
+        {
+            if (dosleep)
+                Thread.sleep(dur!("msecs")( 10 ));
+            else
+                dosleep = true;
+            GC.collect;
```

Comment:
```std.concurrency``` is designed around having a global variable (```scheduler```) set once at the outset. There is no synchronization for this variable and the implementation does not appear to support changing it on the fly.

However we need to test with both implementations of ```Scheduler```: ```ThreadScheduler``` and ```FiberScheduler```.

This function waits until it is the only thread before modifying ```scheduler``` (i.e. it's a mutual exclusion hack.)
Collection helps because threads can wait on the finalization action of other threads (e.g. waiting for ```OwnerTerminated``` exceptions initiated by ```static ~this```.)